### ISSUE TITLE
getThumbByteData (width, height control not work ), return 3000*4000 or 4000*3000 all

### ISSUE THREAD

**Issue by iyuhang** [Sh1d0w/multi_image_picker#438]
**Bug Report**

`multi_image_picker` version:  ^4.6.6+rc.1

Is it happening on Android: yes
Is it happening on iOS: no

```
[✓] Flutter (Channel unknown, v1.12.13+hotfix.5, on Mac OS X 10.14.6 18G87, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.0)
[✓] Connected device (1 available)
```

**Describe the bug**

getThumbByteData (width, height control not work ), return 3000*4000 or 4000*3000 all

![image](https://user-images.githubusercontent.com/14176635/79709632-4980d600-82f5-11ea-8768-177c780892a3.png)

**Comment by iyuhang** [comment]
@Sh1d0w 
Any progress on this issue?
getThumbByteData
width and height control do not make effect

**Comment by longx1001** [comment]
I also encountered this situation, I read the source code, the width or height set is actually the minimum value, the decoder uses a final value based on powers of 2, any other value will be rounded down to the nearest power of 2; for example : The width of the original picture is 1368px, it must be set less than 1368 / (2 * N)), N is an integer greater than 1

**Comment by Sh1d0w** [comment]
Hi guys, Yes that was a change I've made recently that maybe caused a regression. @iyuhang can you please try version v4.6.4 and let me know if it works for you like it should there? If yes I will revert the change I've made.

**Comment by iyuhang** [comment]
Hi @Sh1d0w 
I tried v4.6.4, the issue is still there

**Comment by longx1001** [comment]
> Hi guys, Yes that was a change I've made recently that maybe caused a regression. @iyuhang can you please try version v4.6.4 and let me know if it works for you like it should there? If yes I will revert the change I've made.

Ready to use, thanks

**Comment by longx1001** [comment]
> Hi @Sh1d0w
> I tried v4.6.4, the issue is still there

pubspec.yaml:
```
dependencies:
multi_image_picker: 4.6.4
```
> no ^4.6.4

**Comment by Sh1d0w** [comment]
@iyuhang How do you test this? I just tried and I can not reproduce it on the latest version. How I tested?

I used the example app. Before that I created a sample PHP script that accepts an image and saves it to the computer:

```php
<?php
 
 move_uploaded_file($_FILES["image"]["tmp_name"], 'image.jpeg');
```

Then when I pick the image from the picker I send it to that script via POST request. I am picking the image with size 7500x4000 and I am resizing it to 100x100:

```
Uri uri = Uri.parse('http://test.test/index.php');

// create multipart request
    MultipartRequest request = MultipartRequest("POST", uri);

    ByteData byteData = await resultList[0].getThumbByteData(100, 100);
    List<int> imageData = byteData.buffer.asUint8List();

    MultipartFile multipartFile = MultipartFile.fromBytes(
      'image',
      imageData,
      filename: 'some-file-name.jpg',
    );

// add file to multipart
    request.files.add(multipartFile);
// send
    var response = await request.send();
```

The result is:

<img width="233" alt="Screenshot 2020-04-30 at 9 26 00" src="https://user-images.githubusercontent.com/5074917/80679121-f103c280-8ac4-11ea-9c27-32628e859fbe.png">

**Comment by iyuhang** [comment]
@Sh1d0w here is my code , I write the data to file, and then upload the file
```
Future<File> writeToFile(ByteData data, String path) {
  final buffer = data.buffer;
  return new File(path)
      .writeAsBytes(buffer.asUint8List(data.offsetInBytes, data.lengthInBytes));
}

chooseImages(
    {int maxImages = 1, bool thumbnail = true, persistent = false}) async {

  List<Asset> resultLists;
  List<ChooseMedia> chooseMedias = [];
  try {
    resultLists = await MultiImagePicker.pickImages(
        maxImages: maxImages,
        enableCamera: true,
        materialOptions: MaterialOptions(
            statusBarColor: '#000000',
            actionBarColor: '#000000',
            actionBarTitleColor: '#ffffff',
            useDetailsView: true,
            startInAllView: Platform.isIOS ? true : false,
            allViewTitle: "全部图片",
            lightStatusBar: true),
        cupertinoOptions: CupertinoOptions(backgroundColor: '#ffffff'));
  } catch (e) {
    log(e);
    return List<ChooseMedia>();
  }

  for (int i = 0; i < resultLists.length; i++) {
    final paths = join(
        (await (persistent
                ? getApplicationDocumentsDirectory()
                : getTemporaryDirectory()))
            .path,
        resultLists[i].name);

    File image;

    var bytedata;

    if (thumbnail) {
      int width = resultLists[i].originalWidth;
      int height = resultLists[i].originalHeight;
      log(resultLists[i].name);

      int thumbWidth;

      if (width > height) {
        thumbWidth = 2560;
      } else if (height > width) {
        thumbWidth = 2048;
      } else {
        thumbWidth = 2048;
      }

      bytedata = await resultLists[i].getThumbByteData(
          thumbWidth, (thumbWidth * height ~/ width).toInt(),
          quality: Platform.isIOS ? 95 : 95);
    } else {
      bytedata = await resultLists[i].getByteData();
    }

    await writeToFile(bytedata, paths);

    image = File(paths);

    ChooseMedia media = ChooseMedia(
        mediaType: 0, 
        localFile: image,
        path: paths,
        type: "localImage",
        width: resultLists[i].originalWidth,
        height: resultLists[i].originalHeight);
    chooseMedias.add(media);
  }
  return chooseMedias;
}
```

```
Future uploadFile(File file) async {
  //上传
  Map data = {'bucket': 'loveindex'};
  Map<String, dynamic> res = await Request().post(data, '/account/qiniu_token/');
  log('request qiniu token $res');
  FormData formData = FormData.fromMap({
    "key": Uuid().v1(),
    "token": res['uptoken'],
    "file": await MultipartFile.fromFile(file.path, filename: 'upload_picture')
  });
  Response response = await Dio().post("https://up.qbox.me", data: formData);
  log("qiniu response is $response"); //{"hash":"FtPg_Xu2vkWVUrI5QE2vQudk1toh","key":"1557307812183"}

  //上传完成的 url
  String url = 'http://image.fixtime.com/' + jsonDecode(response.toString())['key'];
  
  return {"url":url};
}
```

**Comment by Sh1d0w** [comment]
```
if (thumbnail) {
      int width = resultLists[i].originalWidth;
      int height = resultLists[i].originalHeight;
      log(resultLists[i].name);

      int thumbWidth;

      if (width > height) {
        thumbWidth = 2560;
      } else if (height > width) {
        thumbWidth = 2048;
      } else {
        thumbWidth = 2048;
      }

      bytedata = await resultLists[i].getThumbByteData(
          thumbWidth, (thumbWidth * height ~/ width).toInt(),
          quality: Platform.isIOS ? 95 : 95);
    } else {
      bytedata = await resultLists[i].getByteData();
    }
```

Are you sure you are not entering into the `else` condition? getByteData returns the full sized image. Can you please remove the whole if / else block and hardcode just 

```
bytedata = await resultLists[i].getThumbByteData(
          200, 200,
          quality: Platform.isIOS ? 95 : 95);
```
Just to confirm it is not a problem with your logic?

**Comment by iyuhang** [comment]
@Sh1d0w 

I changed to hardcode, and if set as follow:

```
bytedata = await resultLists[i].getThumbByteData(
          200, 200,
          quality: Platform.isIOS ? 95 : 95);
```
returns 
![image](https://user-images.githubusercontent.com/14176635/80680981-58843700-8af2-11ea-8e06-c234a1666ba2.png)

but if set to size 2048
```
bytedata = await resultLists[i].getThumbByteData(
          2048, 2048,
          quality: Platform.isIOS ? 95 : 95);
```
returns 
![image](https://user-images.githubusercontent.com/14176635/80681034-705bbb00-8af2-11ea-98be-fac8c388917f.png)

**Comment by longx1001** [comment]
@iyuhang 

你需要按照我的操作更新到指定版本，然后stop后重新编译budeg版本调试
flutter run --debug
https://github.com/Sh1d0w/multi_image_picker/issues/438#issuecomment-621637589

**Comment by iyuhang** [comment]
@longx1001 
yes, if use 
`multi_image_picker: 4.6.4`
there is no issue

**Comment by Sh1d0w** [comment]
You are right, it happens on Android I was testing on iOS. It was actually regression so I've reverted it and released version 4.6.7 which should work as before.

Thanks for the patience!
